misc/python/materialize/mzcompose/services.py

```diff
@@ -608,6 +608,8 @@ def __init__(
         name: str = "testdrive",
         mzbuild: str = "testdrive",
         materialize_url: str = "postgres://materialize@materialized:6875",
+        materialize_username: Optional[str] = None,
+        materialize_password: Optional[str] = None,
```

Comment:
We should keep the options as orthogonal as possible! Here and in testdrive itself. `materialize_url` already lets you set the username/password as you need! `postgres://user:pass@materialized:6875`. Can we just configure that whole string where we need to? And maybe always pass  `materialized_url` as an env var to testdrive below if it otherwise would get leaked in a log file somewhere?

Reply:
No, unfortunately tokio_posgres is unable to handle usernames that contain a `@` if they are passed as part of the URL. It thinks that `@` is the separator between the username and the hostname and it all goes downhill. At the same time, tokio_postgres does not handle URL-style escaping, so no matter how much I sliced it and diced it I could not get it to work.